b2sdk/api.py

```diff
@@ -208,6 +215,12 @@ def create_bucket(
                                              than requested: %s != %s' % (
             bucket_type, bucket.type_
         )
+        if default_retention is not None:
```

Comment:
I'm afraid this could cause very confusing errors, as described in out conversation about retention settings for files. If the key used to authenticate is allowed to create buckets, but does not have `writeBucketRetentions` then creating a bucket will fail, but the bucket will be created (and the name taken) - and that's confusing.